pydropsonde/processor.py

```diff
@@ -659,23 +652,28 @@ def get_flight_attributes(
         """
         flight_attrs = {}
 
-        with open(self.afile, "r") as f:
-            lines = f.readlines()
-
-        for attr in l2_flight_attributes_map.keys():
-            for line_id, line in enumerate(lines):
-                if attr in line:
+        if not self.afile:
+            print(
+                f"No flight attributes for sonde {self.serial_id} on {self.flight_id}"
+            )
+        else:
+            with open(self.afile, "r") as f:
+                lines = f.readlines()
+
+            for attr in l2_flight_attributes_map.keys():
+                for line_id, line in enumerate(lines):
+                    if attr in line:
+                        break
+
+                attr = l2_flight_attributes_map.get(attr, attr)
+                try:
+                    value = lines[line_id].split("= ")[1]
+                    flight_attrs[attr] = float(value) if "AVAPS" not in attr else value
+                except UnboundLocalError:
```

Comment:
there's still an `UnboundLocalError`, because of the use of `line_id` from the inside of the for loop.